_tasks/util/gh-file-importer.mjs

```diff
@@ -0,0 +1,174 @@
+        destDir,
+        ['string'],
+        'a destination directory must be supplied'
+      );
+    }
+    this._log = log;
+    // We need an API key since the GitHub API throttling eventually gets you.
+    // TODO(DerekNonGeneric): Add README for this task w/ guidance on acquiring.
+    if (process.env.GITHUB_SECRET) {
+      // Insert a GitHub token (hope to have one at this point since README).
+      this._octokit = new Octokit({ auth: `${process.env.GITHUB_SECRET}` });
+    }
+    // TODO(DerekNonGeneric): Else automatically get a new GitHub token.
+    // @see https://hg.mozilla.org/mozilla-central/file/tip/tools/github-sync/sync-to-github.sh#l54
+
+    this._destDir = destDir;
+  }
+
+  /**
+   * Validates a request to retrieve a path's metadata prior to doing so.
+   * @param {string} owner The username associated with the repository.
+   * @param {string} repo The repository's name.
+   * @param {!(string | undefined)} path Path to the file or folder.
+   * @param {!(string | undefined)} ref The name of the commit/branch/tag.
+   * @returns {Promise<Object>} An object containing all information.
+   * @public @static @memberof GhFileImporter
+   */
+  async fetchPathMetadata(owner, repo, path = undefined, ref = undefined) {
+    const args = Array.from(arguments);
+    // log.debug({ args });
+    const argNames = ['owner', 'repo', 'path', 'ref'];
+    const hashmap = new Map();
+
+    // Refinement may be necessary depending on whether arguments were omitted.
+    args.forEach(function (value, index) {
+      switch (argNames[index]) {
+        case 'owner':
+          // log.debug('arg case: ' + argNames[index]);
+          if (value === undefined) {
+            // `owner` and/or `repo` are missing, so we throw an error.
+            throw new InvalidArgTypeError(argNames[index], ['string']);
+          }
+          hashmap.set(argNames[index], value);
+          break;
+        case 'repo':
+          // log.debug('arg case: ' + argNames[index]);
+          if (value === undefined) {
+            // `owner` and/or `repo` are missing, so we throw an error.
+            throw new InvalidArgTypeError(argNames[index], ['string']);
+          }
+          hashmap.set(argNames[index], value);
+          break;
+        case 'path':
+          // log.debug('arg case: ' + argNames[index]);
+          if (value === undefined) {
+            log.debug(
+              `${curlyQuote('path')} is missing, so we omit it ` +
+                'causing Octokit to use the repo root directory.'
+            );
+          } else {
+            hashmap.set(argNames[index], value);
+          }
+          break;
+        case 'ref':
+          // log.debug('arg case: ' + argNames[index]);
+          if (value === undefined) {
+            log.debug(
+              `${curlyQuote('ref')} is missing, so we omit it ` +
+                'causing Octokit to use the repo default branch (usually ' +
+                `${curlyQuote('master')}).`
```

Comment:
```suggestion
                `${curlyQuote('main')}).`
```
by the end of the year, this will be the default for all new github repos, so it's probably best to use this one